### PR TITLE
Add Response Convenience

### PR DESCRIPTION
**Pull request by Jonas-Isr**
## Context

AI/ai-sdk-java-backlog#110.

As an orchestration user, I want to have a convenient way of accessing the LLM result and further related information.

### Feature scope:
 
- [x] Added methods `getTokenUsage()` and `getAllMessages()` to `OrchestrationChatResponse`

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [ ] ~Error handling created / updated & covered by the tests above~
- [x] Aligned changes with the JavaScript SDK
- [ ] ~Documentation updated~
- [ ] ~Release notes updated~

### CODE REVIEW

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java [hunk 1]
```diff
@@ -38,4 +42,24 @@ public String getContent() throws OrchestrationClientException {
...
+    return ((LLMModuleResultSynchronous) originalResponse.getOrchestrationResult()).getUsage();
+  }
+
+  /**
+   * Get all messages.
+   *
+   * @return A list of all messages.
+   */
+  @Nullable
+  public List<ChatMessage> getAllMessages() {
```
Comment (CharlesDuboisSAP) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java:62]:
```suggestion
  public List<ChatMessage> getMessageHistory() {
```
debatable @MatKuhr @newtork

**Review by CharlesDuboisSAP — APPROVED** [review]

**Review by CharlesDuboisSAP — DISMISSED** [review]

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java [hunk 2]
```diff
@@ -38,4 +41,29 @@ public String getContent() throws OrchestrationClientException {
...
+   * @return A list of all messages.
+   */
+  @Nonnull
+  public List<ChatMessage> getAllMessages() {
+    final var allMessages = originalResponse.getModuleResults().getTemplating();
+
+    if (allMessages == null) {
+      return List.of();
+    }
+    return allMessages;
```
Comment (MatKuhr) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java:67]: this is missing the assistant response

**Review by MatKuhr — CHANGES_REQUESTED** [review]

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java (hunk 1 above)
Comment (Jonas-Isr) [reply orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java:55]: Kept "getAllMessages".

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java [hunk 3]
```diff
@@ -24,18 +28,46 @@ public class OrchestrationChatResponse {
...
+   * Get list of choices.
+   *
+   * @return A list of choices.
+   */
+  @Nonnull
+  private LLMChoice getCurrentChoice() {
+    return ((LLMModuleResultSynchronous) originalResponse.getOrchestrationResult())
+        .getChoices()
+        .get(0);
+  }
```
Comment (newtork) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java:72]: 1. Incorrect javadoc
2. what happened to null checks?

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java [hunk 4]
```diff
@@ -24,18 +28,46 @@ public class OrchestrationChatResponse {
...
+
+  /**
+   * Get all messages. This can be used for subsequent prompts as a message history.
+   *
+   * @return A list of all messages.
+   */
+  @Nonnull
+  public List<ChatMessage> getAllMessages() {
+    final var messages = originalResponse.getModuleResults().getTemplating();
```
Comment (MatKuhr) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java:56]:
```suggestion
    final var messages = new ArrayList<>(originalResponse.getModuleResults().getTemplating());
```

Otherwise, the subsequent `.add` will manipulate the original response. Repeated calls to `getAllMessages()` thus start to contain the assistant message multiple times. Unfortunately, this is currently missed by the test assertions. Consider adding `assertThat(result.getAllMessages()).hasSize(2);` in the unit test below

**Review by MatKuhr — CHANGES_REQUESTED** [review]
otherwise lgtm

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java (hunk 3 above)
Comment (MatKuhr) [reply orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java:72]: No null check needed IMO because the server should never return empty choices. If it does, throwing an NPE is fine for that case IMO, since we in general don't validate the server response.

File: orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java [hunk 5]
```diff
@@ -380,17 +380,4 @@ void testErrorHandling() {
 
     softly.assertAll();
   }
-
-  @Test
-  void testEmptyChoicesResponse() {
```
Comment (newtork) [inline orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java:385]: (Comment)

I understand it doesn't make sense to cover edge-cases with unit tests.
I just feel defensive about removing test code.

Comment (MatKuhr) [reply orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java:385]: Yeah, in this case I was worried that people later look at this test case and incorrectly conclude this is a valid scenario that might happen, because it is tested

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java (hunk 3 above)
Comment (Jonas-Isr) [reply orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationChatResponse.java]: As discussed, solved by adding a comment in the code.

**Review by MatKuhr — APPROVED** [review]
